test/spawn.jl

```diff
@@ -536,3 +536,77 @@ let
     output = readchomp(pipeline(cmd, stderr=catcmd))
     @test occursin("Info: Hello", output)
 end
+
+# Sys.which() testing
+withenv("PATH" => Sys.BINDIR) do
+    julia_exe = abspath(joinpath(Sys.BINDIR, "julia"))
+    @static if Sys.iswindows()
+        julia_exe *= ".exe"
+    end
+
+    @test Sys.which("julia") == julia_exe
+    @test Sys.which(julia_exe) == julia_exe
+end
+
+mktempdir() do dir
+    withenv("PATH" => dir) do
+        # Test that files lacking executable permissions fail Sys.which
+        # but only on non-Windows systems, as Windows doesn't care...
+        foo_path = abspath(joinpath(dir, "foo"))
+        touch(foo_path)
+        chmod(foo_path, 0o777)
+        @static if !Sys.iswindows()
+            @test Sys.which("foo") == foo_path
+            @test Sys.which(foo_path) == foo_path
+
+            chmod(foo_path, 0o666)
+            @test_throws ErrorException Sys.which("foo")
+            @test_throws ErrorException Sys.which(foo_path)
+        end
+
+        # Test that completely missing files also fail
+        @test_throws ErrorException Sys.which("this_is_not_a_command")
+    end
+end
+
+mktempdir() do dir
+    pathsep = @static if Sys.iswindows() ";" else ":" end
```

Comment:
A ternary would be less weird here

Reply:
`pathsep = Sys.iswindows() ? ";" : ":"` is too much punctuation for me.